source/common/stream_info/stream_info_impl.h

```diff
@@ -271,7 +281,7 @@ struct StreamInfoImpl : public StreamInfo {
   absl::optional<MonotonicTime> last_downstream_tx_byte_sent_;
   absl::optional<MonotonicTime> final_time_;
 
-  absl::optional<Http::Protocol> protocol_;
+  std::vector<std::string> protocols_;
```

Comment:
I would recommend an InlinedVector here with a small number of protocols.

Reply:
Thanks. I am relying on inlined strings to save some space.